Lexer FSM/main.py

```diff
@@ -6,39 +6,74 @@
+            # (not just reject or false) so we can tell if inputCharTerminatesToken
+            if inputCharTerminatesToken and state == accepting:
+                token = determineToken()
+                print(f"Token : {token}, Lexeme: {lexeme}") # can print here or maybe append to list [(token, lexeme), etc.] to print after?
+                lexeme = ""
+                inputCharTerminatesToken = False
+            else:
+                validate_operator(char)
+                validate_separator(char)
+                validate_keyword(char)
+                identifierState = identifier_fsm.validate_identifier(char, prevState)
+                realState = real_fsm.validate_real(char, prevState)
+                integerState = integer_fsm.validate_integer(char, prevState)
+
+                # maybe for this we return 2 values to help us determine if token was terminated (token terminated when we are at accepting state in prevState and then we go to nonaccepting?)
+                integerState, thingToHelpDetermineTerminatedToken = integer_fsm.validate_integer(char, prevState)
+
+                # logic hear using the returned states that will let us know if the token was terminated.
+                # ^ update inputuCharTerminatesToken
+                if thingToHelpDetermineTerminatedToken:
+                    inputCharTerminatesToken = True
+        
+
+
         
-    # Test the Lexer
-    test_case = ["while", "x", "!=", "9", "123.13", "["]
-    for string in test_case:
-        print(lexer(string))
 
 if __name__ == "__main__":
     main()
+
+
+# have a lexer for each type that iterate over the file together. have them each clasify the token as its own type or not its type -- add to a list for each type.
+# then print out at the end in the order that the chars come in
+def identifierLexer(char):
+        lexeme = ""
+        inputCharTerminatesToken = False
+        for char in content:
+            
+            # if we are in one of the finite state machines and we get to a separator, then we should return a value we can identify from the validate_x functions
+            # (not just reject or false) so we can tell if inputCharTerminatesToken
+            if inputCharTerminatesToken and state == accepting: # maybe state will just be true or false so we don't have to do == accepting
+                token = determineToken()
+                print(f"Token : {token}, Lexeme: {lexeme}") # can print here or maybe append to list [(token, lexeme), etc.] to print after?
+                lexeme = ""
+                inputCharTerminatesToken = False
+            else:
+                state = identifier_fsm.validate_identifier(char, prevState)
+                
+
+                # logic hear using the returned states that will let us know if the token was terminated.
+                # ^ update inputuCharTerminatesToken
+                if something:
```

Comment:
?

Reply:
I forgot to change this part to mirror the lexer function up above. it would be in the form of:
 if thingToHelpDetermineTerminatedToken:
                    inputCharTerminatesToken = True

Reply:
ahh okay makes sense